### DISCUSSION TITLE
Better terminology for Airflow's execution start and end date

### DISCUSSION THREAD

**Discussion by howardyoo** [apache/airflow#21130] [General] [answered]
### Apache Airflow version

2.2.3 (latest released)

### What happened

In the future releases, the terminology around how Airflow defines start and end of its `execution` **should be more clearly defined to avoid constant confusion**. I would like to suggest completely replacing the term with `data interval` for better clarity of what it refers to, in the documentation and source code of airflow.

As mentioned in https://stackoverflow.com/questions/67251814/airflow-execution-date-is-confusinghttps://stackoverflow.com/
questions/67251814/airflow-execution-date-is-confusing, the term 'execution' is confusing since many users assume the term `execution` as when the DAG is `executing.` Obviously this has been a constant misunderstanding as execution date simply refers to the period of which the DAG run is going to be basing its processing on.

There is a new term to remedy the `execution` by replacing it with `logical date.` However, the term `logical` is also ambiguous as users may not know the context of which why this is supposed to be `logical.` Therefore, I would like to raise a feature request (if there hasn't been any as there yet) to come up with a better terminology to describe it.

I saw in the Airflow UI which describes the execution date as `data interval` as such:
![image](https://user-images.githubusercontent.com/32691630/151105870-83f51650-d134-475a-9d80-8825b2fa0246.png)

Which, in my opinion, is a far better and accurate term to describe the period of which the DAG run would be basing its processing against. Therefore, I would like to have the existing term `execution date` be deprecated, and gradually be replaced by `data interval` for better clarity.

### What you expected to happen

the term `execution` to describe the period of which the DAG is going to run its orchestration against, is often misleading and confusing. Users would typically think the `execution` start and end date is DAG's run start and end date, and would often get confused when they see different dates on it execution and the DAG's start of the run.

### How to reproduce

Ask any airflow users about what they think about what airflow's execution date means. You will hear many incorrect description of it. If you also talk about it with experienced airflow users, they will often criticize how it is used.

### Operating System

MacOS Monterey (Version 12.1)

### Deployment

Other

### Anything else

This problems occurs constantly among the users. It also happens more to the first time users who are learning airflow.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment, marked as answer]
I converted it into discussion. The `execution_date` is already deprecated. The `logical_date` have been proposed and voted on devlist: 
Here: https://lists.apache.org/thread/16t6mo5r8oj9gyo5zt5rg0pphgjdzd0f

This is such an important decision that it must be discussed on devlist and require voting IMHO. Feel free to either continue the discusion thread I linked to or start a new thread on the devlist, but I see no way any discussion here might help

Apache's "What did not happen on the devlist, did not happen" - is aplicable in this case

**Reply by potiuk** [reply]
BTW. Just to explain - I am not being rude :). 

The reason why it has to be devlist, because we do not want to exclude anyone in the community from the discussion. 
The devlist is "the" most important place where anything really important for Apache projects should happen. This is how Apache Airflow community works (same as any other Apache project: ) https://www.apache.org/foundation/how-it-works.html#communication

**Reply by howardyoo** [reply]
Ok, sorry that I was kind of new into Apache projects and did not know about the devlist. Yes, it makes sense. So, @potiuk , in that case, has it been decided already for the project to move forward with the term `logical date,` and there is not further discussions to be made in this topic?

I was wondering if I could still participate in the discussion in the devlist.

**Reply by potiuk** [reply]
Yes you can. https://airflow.apache.org/community/ here are instructions about joining. You can send message with new thread referring to the other I pointed.

**Reply by howardyoo** [reply]
Thanks, I commented on the thread about why I don't like the term `logical date` to replace `exeuction date`, and my preference to `data date`.

**Comment by howardyoo** [comment]
Sure, no problem.
How can I participate into this dev list?
I do have some opinion about 'logical date' which I... kind of do not like
it.
Like, if someone hear about Logical date..., will he/she understand what it
is right away? For me, I was even more confused because 'logica' could hold
many meanings.

Howard
